src/jit/compiler.cpp

```diff
@@ -2440,147 +2484,111 @@ void Compiler::compSetProcessor()
 
     if (!jitFlags.IsSet(JitFlags::JIT_FLAG_PREJIT))
     {
-        if (configEnableISA(InstructionSet_Base))
+        if (configEnableIsa(jitFlags, InstructionSet_Base))
         {
             opts.setSupportedISA(InstructionSet_Base);
-        }
-        if (configEnableISA(InstructionSet_SSE))
-        {
-            opts.setSupportedISA(InstructionSet_SSE);
+
+            if (configEnableIsa(jitFlags, InstructionSet_SSE))
+            {
+                opts.setSupportedISA(InstructionSet_SSE);
 #ifdef _TARGET_AMD64_
-            opts.setSupportedISA(InstructionSet_SSE_X64);
-#endif
-        }
-        if (configEnableISA(InstructionSet_SSE2))
-        {
-            opts.setSupportedISA(InstructionSet_SSE2);
+                opts.setSupportedISA(InstructionSet_SSE_X64);
+#endif // _TARGET_AMD64_
+
+                if (configEnableIsa(jitFlags, InstructionSet_SSE2))
+                {
+                    opts.setSupportedISA(InstructionSet_SSE2);
 #ifdef _TARGET_AMD64_
-            opts.setSupportedISA(InstructionSet_SSE2_X64);
-#endif
-        }
-        if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_LZCNT))
-        {
-            if (configEnableISA(InstructionSet_LZCNT))
-            {
-                opts.setSupportedISA(InstructionSet_LZCNT);
+                    opts.setSupportedISA(InstructionSet_SSE2_X64);
+#endif // _TARGET_AMD64_
+
+                    if (configEnableIsa(jitFlags, InstructionSet_AES))
+                    {
+                        opts.setSupportedISA(InstructionSet_AES);
+                    }
+
+                    if (configEnableIsa(jitFlags, InstructionSet_PCLMULQDQ))
+                    {
+                        opts.setSupportedISA(InstructionSet_PCLMULQDQ);
+                    }
+
+                    // We need to additionaly check that COMPlus_EnableSSE3_4 is set, as that
+                    // is a prexisting config flag that controls the SSE3+ ISAs
+                    if (configEnableIsa(jitFlags, InstructionSet_SSE3) && JitConfig.EnableSSE3_4())
+                    {
+                        opts.setSupportedISA(InstructionSet_SSE3);
+
+                        if (configEnableIsa(jitFlags, InstructionSet_SSSE3))
+                        {
+                            opts.setSupportedISA(InstructionSet_SSSE3);
+
+                            if (configEnableIsa(jitFlags, InstructionSet_SSE41))
+                            {
+                                opts.setSupportedISA(InstructionSet_SSE41);
 #ifdef _TARGET_AMD64_
-                opts.setSupportedISA(InstructionSet_LZCNT_X64);
-#endif
-            }
-        }
-        if (jitFlags.IsSet(JitFlags::JIT_FLAG_USE_POPCNT))
```

Comment:
`POPCNT` in particular was wrong, because it wasn't also respecting `EnableSSE3_4`